### ISSUE TITLE
Observatory profiler doesn't seem to work

### ISSUE THREAD

**Issue by Hixie** [flutter/flutter#35140]
Steps to reproduce:

1. Check out tip-of-tree flutter.
2. Edit `//packages/flutter/test/material/about_test.dart` to add `while (true)` on line 121 (just before a call to pumpWidget). This is just so that the test will run forever. The details don't matter.
3. Run `flutter test --start-paused test/material/about_test.dart`. It will print out a URL.
4. Open the given URL.
5. Click the "Isolate" link at the bottom.
6. Go to main > cpu profile at the top.
7. Click "Enable profiler".
8. Go to the debugger and run "continue".
9. Go back to the profiler and click Refresh. There are no samples.

**Comment by jacob314** [comment]
Fyi @bkonyi @kenzieschmoll 
We're hoping to soon have complete enough support in the DevTools profiler that the answer is to remove the profiler from the observatory not spend time tracking down why it is broken.

**Comment by a-siva** [comment]
if the underlying vm-service protocol is broken it doesn't matter if observatory or devtools is used, we have to track it down and fix.

**Comment by a-siva** [comment]
It appears that a number of samples are being skipped because the register check is failing
single_frame_sample_register_check: 1082

**Comment by cbracken** [comment]
@rmacnak-google @flar I believe the two of you were looking at this (or something similar) on Friday -- was this related or another issue altogether?

**Comment by flar** [comment]
I don't recognize these issues. I had trouble saving a trace from Observatory, but I was able to load data into Observatory just fine.

**Comment by bkonyi** [comment]
> I don't recognize these issues. I had trouble saving a trace from Observatory, but I was able to load data into Observatory just fine.

The issue @flar is seeing is tracked by https://github.com/flutter/flutter/issues/41746 and should be resolved shortly. As for the lack of samples, I've never seen that behavior before.

**Comment by bkonyi** [comment]
It looks like this is due to how `dart:io` handles spawning new processes. On POSIX platforms we block all signals while spawning a process and the new process inherits that state. The standalone embedder and threads spawned by the VM know about this and automatically unblock the relevant signals, but `flutter_tester` doesn't, so `SIGPROF` is blocked for the main isolate's mutator thread. All we have to do is unblock SIGPROF when `flutter_tester` starts and that should resolve this issue.

**Comment by wpietri** [comment]
I think this is the bug I am currently experiencing. Just to check my understanding of the discussion: right now it's not possible to profile a flutter unit test?

**Comment by bkonyi** [comment]
> I think this is the bug I am currently experiencing. Just to check my understanding of the discussion: right now it's not possible to profile a flutter unit test?

That's correct. Once the fix propagates through to the framework you should be able to profile unit tests correctly.
